### ISSUE TITLE
NifUntaggedEnum codegen fails when generic types skip :: syntax

### ISSUE THREAD

**Issue by iredelmeier** [rusterlium/rustler#370]
Given the code:

```rust
#[derive(rustler::NifUntaggedEnum)]
pub enum Example {
    Vec(Vec<usize>),
}
```
you currently get the following compilation errors:

```
error: comparison operators cannot be chained
 --> src/lib.rs:3:12
  |
3 |     Vec(Vec<usize>),
  |            ^     ^
  |
help: use `::<...>` instead of `<...>` to specify type or const arguments
  |
3 |     Vec(Vec::<usize>),
  |            ^^

error: proc-macro derive produced unparseable tokens
 --> src/lib.rs:1:10
  |
1 | #[derive(rustler::NifUntaggedEnum)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^
```

Adding the `::` as suggested resolves the problem, although this isn't really idiomatic and gets reverted by rustfmt (at least by default).

Wrapping the type also successfully circumvents it.

**Comment by evnu** [comment]
The proc-macro generates the following `Decoder` implementation:

```rust
impl<'a> ::rustler::Decoder<'a> for Example {
    fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
        if let Ok(inner) = () {
            return Ok(Example::Vec(inner));
        }
        Err(::rustler::Error::Atom("invalid_variant"))
    }
}
```

When using `Vec::<usize>`, the expansion works:

```rust
impl<'a> ::rustler::Decoder<'a> for Example {
    fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
        if let Ok(inner) = Vec::<usize>::decode(term) {
            return Ok(Example::Vec(inner));
        }
        Err(::rustler::Error::Atom("invalid_variant"))
    }
}
```

My guess is [that extracting `field_type`](https://github.com/rusterlium/rustler/blob/master/rustler_codegen/src/untagged_enum.rs#L57) does not work as expected:

```rust
            let field_type = &variant.fields.iter().next().unwrap().ty;
```

**Comment by evnu** [comment]
Thanks for the report @iredelmeier. The fix is quite simple #371 and is basically taken straight from the readme of [quote](https://github.com/dtolnay/quote#making-method-calls).
